lib/src/instances/stream_monad.dart

```diff
@@ -68,10 +84,109 @@ class StreamMonad<T> extends Monad<T> implements Stream<T> {
   StreamMonad<E> asyncMap<E>(FutureOr convert(T event)) =>
       new StreamMonad(_stream.asyncMap(convert));
 
+  /// Combines multiple Observables to create an Observable whose values are
+  /// calculated from the latest values of each of its input Observables.
+  StreamMonad<E> combineLatest<S, E>(
+      StreamMonad<S> other, E combine(T element, S otherElement)) {
```

Comment:
Missed this!
Types don't match, we would need to remove the `E` type parameter and instead replace it for `Tuple2<T, S>` I actually like that alternative, consumers can then compose with the combine function if they need to, thoughts?